py2exe/mf3.py

```diff
@@ -684,6 +691,27 @@ def __dest_file__(self):
             return self.__name__.replace(".", "\\") + bytecode_suffix
 
 
+    @property
+    def __filename__(self):
+        if not hasattr(self, "__file__"):
+            return None
+        assert self.__file__.endswith(self.__extension__)
+        return self.__file__
+
+
+    @property
+    def __extension__(self):
+        if self.__loader__.__class__ != importlib.machinery.ExtensionFileLoader:
+            return None
+        if self.__file__.endswith(EXTENSION_TARGET_SUFFIX):
+            return EXTENSION_TARGET_SUFFIX
+        if 'pywintypes' in self.__file__:

```

Comment:
I am not super convinced by this, it feels like a hack. I did not have the time to investigate why these DLLs are picked in the module and not as external libraries, which is causing this issue in the first place.

Reply:
Ah, the issue that is causing it in the first place is that `determine_dll_type` sees a linkage to the python DLL in the dependencies.  I checked with Dependency Walker and this linkage is real.

Another choice is to check for pywintypes and pywincom in `determine_dll_type` and treat them as a regular DLL, but my concern with that approach is that py2exe would not build a wrapper such that "import" would work properly on those modules:
https://github.com/albertosottile/py2exe/blob/master/py2exe/runtime.py#L443